pkg/cosign/cosign.go

```diff
@@ -146,14 +146,22 @@ func VerifySignature(opts Options) (digest string, err error) {
 		return "", errors.Wrap(err, "failed to get payload")
 	}
 
-	issuer, err := extractIssuer(opts.ImageRef, payload, log)
-	if err == nil && (issuer != opts.Issuer) {
-		return "", errors.Wrap(err, "issuer mismatch")
+	if opts.Issuer != "" {
+		issuer, err := extractIssuer(opts.ImageRef, payload, log)
```

Comment:
Does an image signed with public/private key-pair have `subject` and `issuer` fields in their payload?

If they don't then the extractIssuer func should return an error and the following statement (Line 151) shouldn't be executed.